### ISSUE TITLE
Unpack/pack code order

### ISSUE THREAD

**Issue by fredrikbk** [tensor-compiler/taco#74]
When we emit C code the order of the statements that unpacks and restricts pointers (and packs) are non-deterministic. They should be ordered by order of tensors and indices. That is:
```
  double* restrict a_vals = (double*)(a->vals);
  int a0_size = *(a->indices[0][0]);
  int B0_size = *(B->indices[0][0]);
  int* restrict B1_idx = (int*)(B->indices[1][1]);
  int c0_size = *(c->indices[0][0]);
  double* restrict B_vals = (double*)(B->vals);
  double* restrict c_vals = (double*)(c->vals);
  int* restrict B1_pos = (int*)(B->indices[1][0]);
```
should be:
```
  int a0_size = *(a->indices[0][0]);
  double* restrict a_vals = (double*)(a->vals);
  int B0_size = *(B->indices[0][0]);
  int* restrict B1_pos = (int*)(B->indices[1][0]);
  int* restrict B1_idx = (int*)(B->indices[1][1]);
  double* restrict B_vals = (double*)(B->vals);
  int c0_size = *(c->indices[0][0]);
  double* restrict c_vals = (double*)(c->vals);
```

**Comment by shoaibkamil** [comment]
Unpack code order is fixed by 6c58773.

**Comment by shoaibkamil** [comment]
Pack code order is fixed by 44bc3e8.
